src/test/java/echoserver/EchoServerServiceTest.java

```diff
@@ -0,0 +1,130 @@
+package echoserver;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
```

Comment:
One thing I like about Java is the imports. The imports tell you about the file as a whole. How do you feel about this list of imports for the `EchoServerServiceTest`?

I am wondering if the test know too much about the details of the implementation. There are multiple levels of abstraction present here in the list.